src/main/java/App.java

```diff
@@ -0,0 +1,19 @@
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class App {
+    public static void main(String[] args) throws IOException {
+        UserIO console = new UserIO(System.in, System.out);
```

Comment:
This method is very busy. There are a lot of responsibilities hre. I like that you create a `UserIO` straight away. Why not do this for client and server apps?